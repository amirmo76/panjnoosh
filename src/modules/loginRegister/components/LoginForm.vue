<template>
  <form @submit="onSubmit">
    <BaseInput :type="'email'" @valueChange="onEmailChange" :error="emailError" :id="'email'">ایمیل</BaseInput>
    <BaseInput
      :type="'password'"
      @valueChange="onPasswordChange"
      :error="passwordError"
      :id="'password'"
    >رمز عبور</BaseInput>
    <BaseButton :stretched="true" class="button">ورود</BaseButton>
  </form>
</template>

<script>
import BaseInput from "../../../core/components/BaseInput";
import BaseButton from "../../../core/components/BaseButton";
import { mapActions } from "vuex";

export default {
  name: "LoginForm",
  components: {
    BaseButton,
    BaseInput
  },
  data() {
    return {
      email: "",
      password: "",
      emailError: "",
      passwordError: ""
    };
  },
  methods: {
    ...mapActions(["login"]),
    onEmailChange(val) {
      const regex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
      if (regex.test(val)) {
        this.email = val;
        this.emailError = "";
      } else {
        this.emailError = "ایمیل وارد شده نامعتبر می باشد";
      }
    },
    onPasswordChange(val) {
      if (val.length > 5) {
        this.passwordError = "";
        this.password = val;
      } else {
        this.passwordError = "رمز عبور باید بیش از ۵ کاراکتر باشد";
      }
    },
    onSubmit(e) {
      e.preventDefault();
      this.login({
        email: this.email,
        password: this.password
      });
    }
  }
};
</script>

<style scoped>
form {
  display: flex;
  flex-direction: column;
  width: 50%;
  justify-content: center;
}

form > *:not(:last-child) {
  margin-bottom: 5rem;
}

.button {
  margin-top: 2rem;
}
</style>


