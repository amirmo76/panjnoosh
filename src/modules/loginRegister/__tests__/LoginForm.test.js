import { shallowMount } from "@vue/test-utils";
import LoginForm from "../components/LoginForm";

test("should match snapshot", () => {
  const wrapper = shallowMount(LoginForm);
  expect(wrapper.element).toMatchSnapshot();
});

test("should validate email correctly", () => {
  const wrapper = shallowMount(LoginForm);
  wrapper.vm.onEmailChange("12324");
  expect(wrapper.vm.email).toBe("");

  wrapper.vm.onEmailChange("amirmo");
  expect(wrapper.vm.email).toBe("");

  wrapper.vm.onEmailChange("amir@gmail.com");
  expect(wrapper.vm.email).toBe("amir@gmail.com");

  wrapper.vm.onEmailChange("");
  expect(wrapper.vm.email).toBe("amir@gmail.com");

  wrapper.vm.onEmailChange("ali@yahoo.com");
  expect(wrapper.vm.email).toBe("ali@yahoo.com");
});

test("should validate passwrod correctly", () => {
  const wrapper = shallowMount(LoginForm);
  wrapper.vm.onPasswordChange("12345");
  expect(wrapper.vm.password).toBe("");

  wrapper.vm.onPasswordChange("0965687");
  expect(wrapper.vm.password).toBe("0965687");
});

// test("should call onEmailChange on email input with the right value", () => {
//   // test
//   const wrapper = mount(LoginForm);
//   const spy = jest.spyOn(wrapper.vm, "onEmailChange");

// });

// test("should call onPasswordChange on password input with the right value", () => {
//   // test
// });

// test("should call onSubmit on clicking submit ", () => {
//   // test
// });
