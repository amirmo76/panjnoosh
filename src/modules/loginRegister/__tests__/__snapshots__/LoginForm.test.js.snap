// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<form>
  <baseinput-stub
    error=""
    id="email"
    type="email"
  >
    ایمیل
  </baseinput-stub>
   
  <baseinput-stub
    error=""
    id="password"
    type="password"
  >
    رمز عبور
  </baseinput-stub>
   
  <basebutton-stub
    class="button"
    stretched="true"
    theme="primary"
    title=""
  >
    ورود
  </basebutton-stub>
</form>
`;
